### DISCUSSION TITLE
Problem with pushing a commit from Practical 4

### DISCUSSION THREAD

**Discussion by cheisbaee** [IBAHCM/RPiR#547] [Q&A for technical issues] [answered]
Hi,

I am having issues pushing any updates that I have to my package to GitHub. 

When I try to push any of my commits through RStudio gives me the message: >>> /usr/bin/git push origin HEAD:refs/heads/main
error: cannot run rpostback-askpass: No such file or directory
fatal: could not read Username for 'https://github.com': terminal prompts disabled

The git terminal also has a message saying, " Your branch is ahead origin/main by 1 commit". 

Is there any way that I can solve this problem? Any solutions would be much appreciated.

**Comment by richardreeve** [comment]
Hi @cheisbaee - have you updated your computer in the last few days? It sounds like parts of git are not installed, which is odd since you have obviously been able to use git in the past. If that might be the problem, then you can try to reinstall it - go to:

https://support.posit.co/hc/en-us/articles/200532077-Version-Control-with-Git-and-SVN

And scroll down to the installation section. Can you push code from the project or one of the other exercises btw? Oh, and what operating system are you using? Windows or Mac?

**Reply by cheisbaee** [reply]
Hi Richard,

I think my computer did update itself. I will try the method you suggested. Unfortunately, I am having the same difficulties when I try to push any code from the previous exercises.

Kind regards,
Che

**Reply by cheisbaee** [reply]
I am using a Mac^

**Reply by richardreeve** [reply]
I think if there was a system update and it doesn't work with other repositories that's good news - following the instructions to reinstall git should work then I hope.

**Comment by cheisbaee** [comment]
Hi Richard,

Having followed the instructions for reinstallation, I reinstalled git by using home-brew for MacOSx. 

However, I still could not resolve the issue. 

Do you know of any other potential causes of this issue and alternative ways that I can fix it?

**Reply by richardreeve** [reply]
I'm happy to have a zoom call if you want to - email me and we can set one up. This kind of technical problem is very tricky to resolve except in person (and even then not so easy!).

**Comment by richardreeve** [comment, marked as answer]
If people have a problem pushing changes to GitHub - particularly if they remember getting a message from GitHub about their Personal Access Token expiring - then it may be worth repeating the instructions for creating a personal access token and saving it in RStudio here:

https://sbohvm.github.io/RPiR/articles/pages/install_github.html#add-a-personal-access-token-pat

The process will be slightly different in RStudio because you are replacing an existing token rather than creating one for the first time, but the differences are minor and should be obvious.
